### DISCUSSION TITLE
"Adding your SSH key to the ssh-agent" is inaccurate: "Host *.github.com" doesn't work

### DISCUSSION THREAD

**Discussion by jnussbaum** [community/community#45792] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

The documentation at https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent is slightly inaccurate, IMHO.

It tells me to add this to my `~/.ssh/config`:
```
Host *.github.com
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile ~/.ssh/id_ed25519
```

Unfortunately, this doesn't work for me. However, if I make it a global rule instead of restricting it to `*.github.com`, then I succeed: 
```
Host *
  ...
```

Background: I migrated my ssh key to my new MacBookPro 16-inch, 2021, Apple M1 Max, Ventura 13.0. My ssh key is protected with a password. I added the password to Keychain Access with `ssh-add`, as described in the docs cited above, so that I don't have to type it in every time. This worked fine, they password is actually stored in Keychain Access.

But still, every time I wanted to clone a repo over SSH, Terminal asked me for my password. I tried out different things, but in the end I found out that the issue is the restriction to `*.github.com`. So I think that the docs must be corrected. 

(My guess is that in the background, there is a call to a URL/subdomain which doesn't match `*.github.com`.)

Any thoughts / corrections?

**Comment by jge162** [comment]
Very detailed post @jnussbaum. You should consider submitting a bug report for the developers.

**Reply by jge162** [reply]
[Bug report link](https://support.github.com/contact/bug-report)

**Reply by jnussbaum** [reply]
Thanks for the suggestion. I opened a bug ticket here: https://support.github.com/ticket/personal/0/1992941
